src/Parser/Shaman/Elemental/Modules/ShamanCore/EarthShock.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import ItemLink from 'common/ItemLink';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatNumber } from 'common/format';
+
+class EarthShock extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+    combatants: Combatants,
+  };
+
+  maelstromSpendOutAscendence = [];
+  maelstromSpendInAscendence = [];
+  latershock = false;
+  gambling = false;
+
+  on_initialized() {
+    this.gambling = (this.combatants.selected.hasHands(ITEMS.SMOLDERING_HEART.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
+    this.pseudogambling = (this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
```

Comment:
Please add this variable also to the list of variables few lines up, makes it easier to read.